src/mapping/FGreedyCutSolver.hpp

```diff
@@ -0,0 +1,243 @@
+#pragma once
+#include <Eigen/Cholesky>
+#include <Eigen/QR>
+#include <Eigen/SVD>
+#include <boost/range/adaptor/indexed.hpp>
+#include <boost/range/irange.hpp>
+#include <numeric>
+#include "io/ExportVTU.hpp"
+#include "mapping/config/MappingConfigurationTypes.hpp"
+#include "mapping/RadialBasisFctSolver.hpp"
+#include "mesh/Mesh.hpp"
+#include "precice/impl/Types.hpp"
+#include "profiling/Event.hpp"
+#include <fenv.h>
+
+namespace precice {
+namespace mapping {
+
+
+template <typename RADIAL_BASIS_FUNCTION_T>
+class FGreedySolver {
+public:
+  using DecompositionType = std::conditional_t<RADIAL_BASIS_FUNCTION_T::isStrictlyPositiveDefinite(), Eigen::LLT<Eigen::MatrixXd>, Eigen::ColPivHouseholderQR<Eigen::MatrixXd>>;
+  using BASIS_FUNCTION_T  = RADIAL_BASIS_FUNCTION_T;
+  /// Default constructor
+  FGreedySolver() = default;
```

Comment:
```suggestion
  FGreedySolver() = default;
```
I think that's not required anymore (it was in the past).